make/langtools/tools/compileproperties/CompileProperties.java

```diff
@@ -183,7 +183,7 @@ private boolean parseOptions(String args[]) {
                         log.error("cannot close " + filename, e);
                     }
                 }
-                if ( ok = true && contents != null ) {
+                if ( ok && contents != null ) {
                     String tokens[] = (new String(contents)).split("\\s+");
```

Comment:
So the intended composite predicate here is thought to be
ok == true && contents != null
which is equivalent to
ok && contents != null.
The semantics of the current code are equivalent to just
contents != null
right?

Reply:
The semantics is definitely changed (considering the first branch was always true originally) but it may be the original implementation being incorrect,

Reply:
well not assigning `true` to `ok` also breaks the semantics as `ok` is used in boolean expressions below this point. So if we want to keep the semantics 100% we will have to either let the code as it was or:
```
ok = true;
if (contents != null) {
   ...
```